apps/src/applab/CompletionButton.jsx

```diff
@@ -0,0 +1,70 @@
+var msg = require('../locale');
+
+var connect = require('react-redux').connect;
+var ProtectedStatefulDiv = require('../templates/ProtectedStatefulDiv');
+
+var styles = {
+  main: {
+    display: 'inline'
+  }
+};
+
+/**
+ * Component for for potentially showing a completion button in applab.
+ * Depending on props, this button (if it exists) will say "Finish", "Submit",
+ * or "Unsubmit"
+ */
+var CompletionButton = React.createClass({
+  propTypes: {
+    isProjectLevel: React.PropTypes.bool.isRequired,
+    isSubmittable: React.PropTypes.bool.isRequired,
+    isSubmitted: React.PropTypes.bool.isRequired,
+  },
+
+  render: function () {
+    var id;
+    var contents;
+    var divClass = 'share-cell-none';
+
+    if (this.props.isProjectLevel) {
+      return <div/>;
```

Comment:
Means the parent needs to know if this isProjectLevel, which it doesn't need to know currently.